src/Kupo/Data/Database.hs

```diff
@@ -149,6 +151,43 @@ patternToSql = \case
     MatchPaymentAndDelegation payment delegation ->
         "LIKE '__" <> encodeBase16 payment <> encodeBase16 delegation <> "'"
 
+
+-- querying whether the given pattern is covered entirely by a pattern stored in the DB
+patternContainsPatterns
+    :: Pattern
+    -> [Text]
+patternContainsPatterns = \case
+    MatchAny IncludingBootstrap ->
+        ["*"]
+    MatchAny OnlyShelley ->
+        "*/*" : (patternContainsPatterns (MatchAny IncludingBootstrap))
+    MatchPayment payment ->
+        encodeBase16 payment <> "/*" : (patternContainsPatterns (MatchAny OnlyShelley))
+    MatchDelegation delegation ->
+        "*/" <> encodeBase16 delegation : (patternContainsPatterns (MatchAny OnlyShelley))
+    MatchPaymentAndDelegation payment delegation ->
+        encodeBase16 payment <> "/" <> encodeBase16 delegation : (
+            patternContainsPatterns (MatchPayment payment) <>
+            patternContainsPatterns (MatchDelegation delegation)
+        )
+    MatchExact addr ->
+        encodeBase16 (Ledger.serialiseAddr addr) :
+        case (getPaymentPartBytes addr) of
+        Nothing ->
+            patternContainsPatterns (MatchAny IncludingBootstrap)
+        Just payment -> 
+             case (getDelegationPartBytes addr) of
+                Nothing ->
+                    patternContainsPatterns (MatchPayment payment)
+                Just delegation -> 
+                    patternContainsPatterns (MatchPaymentAndDelegation payment delegation)
```

Comment:
Rather than working at the database-level, I think we can rework the function to operate over a pair of `Pattern` and returns yes/no if there's any overlap. Fortunately, there already is such a function! 

See https://github.com/CardanoSolutions/kupo/blob/92a64d143accac4081f8e68bedb86684fc2dce07/src/Kupo/Data/Pattern.hs#L85-L110

With a small modification, this function could return the overlapped pattern instead of a boolean. Current call-site can simply check if the result is empty, whereas the new endpoint can leverage this to return overlapping patterns.

Reply:
I have added a function `includes` based on the overlaps function. The rationale is that as a user I would like to make sure that my pattern-frac is completely covered by the given pattern (at least I would want that). I can also imagine adding a ?partial flag that returns overlaps too (or making the full inclusion a flag)

Reply:
I suggest to keep the additional complexity of a relaxing flag out of this PR since neither #40 nor #38 indicate that they would need the option. Unless you have a use case at hand. It can still be added later without problems.

Reply:
> I suggest to keep the additional complexity of a relaxing flag out of this PR 

I concur. And I wouldn't include it anyway unless requested and backed by a use-case.